markets/perps-market/test/integration/Liquidation/Liquidation.margin.test.ts

```diff
@@ -1,38 +1,243 @@
+    await systems().PerpsMarket.connect(trader1()).modifyCollateral(2, 0, bn(20_000));
+  });
+
+  let commonOpenPositionProps: Pick<
+    OpenPositionData,
+    'systems' | 'provider' | 'trader' | 'accountId' | 'keeper'
+  >;
+  before('identify common props', async () => {
+    commonOpenPositionProps = {
+      systems,
+      provider,
+      trader: trader1(),
+      accountId: 2,
+      keeper: trader1(),
+    };
+  });
+
+  before('open positions', async () => {
+    const positionSizes = [
+      bn(1), // btc
+      bn(20), // eth
+      bn(2000), // link
+      bn(5000), // arb
+      bn(5000), // op
+    ];
+
+    for (const [i, perpsMarket] of perpsMarkets().entries()) {
+      await openPosition({
+        ...commonOpenPositionProps,
+        marketId: perpsMarket.marketId(),
+        sizeDelta: positionSizes[i],
+        settlementStrategyId: perpsMarket.strategyId(),
+        price: perpsMarketConfigs[i].price,
+      });
+    }
+  });
+
+  describe('account check after initial positions open', async () => {
+    it('should have correct open interest', async () => {
+      assertBn.equal(await systems().PerpsMarket.totalAccountOpenInterest(2), bn(100_000));
+    });
+
+    [
+      { size: bn(1), pnl: bn(-150) }, // btc
+      { size: bn(20), pnl: bn(-400) }, // eth
+      { size: bn(2000), pnl: bn(-100) }, // link
+      { size: bn(5000), pnl: bn(-250) }, // arb
+      { size: bn(5000), pnl: bn(-250) }, // op
+    ].forEach(({ size, pnl }, i) => {
+      it(`should have correct position for ${perpsMarketConfigs[i].token}`, async () => {
+        const [positionPnl, , positionSize] = await systems().PerpsMarket.getOpenPosition(
+          2,
+          perpsMarkets()[i].marketId()
+        );
+        assertBn.equal(positionPnl, pnl);
+        assertBn.equal(positionSize, size);
+      });
+    });
+
+    it('has correct available margin', async () => {
+      assertBn.equal(await systems().PerpsMarket.getAvailableMargin(2), bn(18_850));
+    });
+  });
+
+  describe('prices changes', () => {
+    [
+      bn(28000), // btc
+      bn(1900), // eth
+      bn(4.6), // link
+      bn(1.9), // arb
+      bn(1.8), // op
+    ].forEach((price, i) => {
+      before(`change ${perpsMarketConfigs[i].token} price`, async () => {
+        await perpsMarkets()[i].aggregator().mockSetCurrentPrice(price);
+      });
+    });
+
+    [
+      bn(-2150), // btc
+      bn(-2400), // eth
+      bn(-900), // link
+      bn(-750), // arb
+      bn(-1250), // op
+    ].forEach((pnl, i) => {
+      it(`should have correct position pnl for ${perpsMarketConfigs[i].token}`, async () => {
+        const [positionPnl] = await systems().PerpsMarket.getOpenPosition(
+          2,
+          perpsMarkets()[i].marketId()
+        );
+        assertBn.equal(positionPnl, pnl);
+      });
+    });
+
+    it('has correct available margin', async () => {
+      assertBn.equal(await systems().PerpsMarket.getAvailableMargin(2), bn(12_550));
```

Comment:
I was gonna comment saying that a comment explaining how we arrive to 12_550 might make sense. 
Then I looked up getAvailableMargin, so it is `totalCollateralValue` + `getAccountPnl`. 

Total collateral value is using a very real life price by looking up what price we would get if we were to sell the collateral. Account pnl on the other hand is using oracle rather than skew adjusted price. I might be wrong but it feels like using fillPrice for pnl would be more accurate represent the true collateral value

Reply:
well account pnl has to do with the perps market price, whereas total collateral value looks at the spot market price (which is related to its own risk mechanism).  we wouldn't necessarily look at spot synth price for a perps market.  hope this makes sense.

Reply:
Yea I didn't mean that we should use the spot market price. I meant the perps skew adjusted price

Reply:
oh yeah, very good point.  in my opinion, even if we use fill price, it's not really an accurate price since when the trader ends up modifying their position, would be average of the current skew price and the skew after the amt they're modifying.  and the price delta gets better/worse depending on how large the size is vs the fill price calculated with skew 0.  

i think i rmr afif mentioning to use on chain price; but it might be good to bring this up during our meetings with afif this week.  i'll take a note and double check 👍🏽 .  thanks for pointing it out.

Reply:
Cool, and yea fill price based on the size of the position would be the way I expect it to work. 

It does ring a bell that we asked afif about this before. I just cant remember any argument for why we wouldn't use fill price.

Reply:
the argument i'd make is i wouldn't think the trader should have a fluctuating liquidation based on the market's skew (fill price is based on market skew).  imagine if you had a long open, and there's a massive long skew.  and i get liquidated even when price didn't change, but skew did when some other trader closed a really large short position; that doesn't seem fair.

Reply:
Hmm yea, that is an argument. Using fill price still seems better to me, that is the current price of the asset in our platform. Just as changing market skew affect the price if I want to close a position I would expect changing market skew to affect liquidation. 

But as long as it gets discussed and it's a conscious decision I'm happy